### ISSUE TITLE
Support SASL authentication

### ISSUE THREAD

**Issue by tacerus** [google/alertmanager-irc-relay#12]
Hi,

Thanks for this nice program!
I would like to submit a feature request: support native SASL authentication (I am aware some IRCd's support SASL through PASS).

Cheers,
Georg

**Comment by shammash** [comment]
Hi Georg,

Thank you for the suggestion.

I skimmed through the SASL specs[1]. Before we can add this feature to the bot we'd have to add capability negotiation and SASL support to the underlying IRC library: http://github.com/fluffle/goirc .

Looping in @fluffle to make sure that's ok.

[1] https://ircv3.net/specs/extensions/sasl-3.1 and https://ircv3.net/specs/extensions/sasl-3.2

Thank you,

Luca

**Comment by fluffle** [comment]
Sure, that'd be great! There's a whole pile of IRCv3 things that would be nice to have. Because IARC I am not sure I can use work time to help out, though. I guess I could ask them what to do about this.
